### PR TITLE
Add examples to Socket::Server#accept API docs

### PR DESCRIPTION
**Pull request by straight-shoota**

### CODE REVIEW

File: src/socket/server.cr [hunk 1]
```diff
@@ -2,12 +2,32 @@ class Socket
...
 
     # Accepts an incoming connection and returns the client socket.
     #
     # Returns `nil` if the server is closed after invoking this method.
+    #
+    # ```
+    # require "socket"
+    #
+    # server = TCPServer.new(2202)
+    # while socket = server.accept
```
Comment (asterite) [inline src/socket/server.cr]: `accept` -> `accept?`

File: src/socket/server.cr [hunk 2]
```diff
@@ -2,12 +2,32 @@ class Socket
   module Server
     # Accepts an incoming connection and returns the client socket.
     #
+    # ```
+    # require "socket"
+    #
+    # server = TCPServer.new(2202)
+    # while socket = server.accept
```
Comment (asterite) [inline src/socket/server.cr]: But `server.accept` is always truthy...

File: src/socket/server.cr [hunk 3]
```diff
@@ -2,12 +2,33 @@ class Socket
   module Server
     # Accepts an incoming connection and returns the client socket.
     #
+    # ```
+    # require "socket"
+    #
+    # server = TCPServer.new(2202)
+    # while true
```
Comment (jhass) [inline src/socket/server.cr]: `loop do`?

Comment (asterite) [reply src/socket/server.cr]: Both are fine. I'm fact, I'm starting to think we should remove `loop`, it adds absolutely nothing and it's just another way of doing something.

Comment (jhass) [reply src/socket/server.cr]: well I guess it was useful when it had its block argument :P

**Pull request merged**